instrumentation/elasticsearch/lib/opentelemetry/instrumentation/elasticsearch/wildcard_pattern.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module Elasticsearch
+      # Object representing a user-supplied key pattern that behaves as a regex
+      class WildcardPattern
```

Comment:
Why not just allow a regex in this case? Is this mimicking an upstream Elasticsearch wildcard pattern? (And, if so, could we just borrow their implementation?)

Reply:
The reason I used a special object for this and didn't require a regex was to allow the config option `sanitize_field_names` to be set as a list of strings or a string representing a list. So the config option could be set as:
`{ sanitize_field_names: 'foo.*,*.bar' } }`
or
`{ sanitize_field_names: ['Auth*tion', 'abc*', '*xyz'] }`

I decided to do that instead of requiring a regex or list of regexes so that the config could be set via an environment variable, if needed. Maybe that's not a valid use case in OTel instrumentations? I'm happy to change this, just let me know what makes more sense for your users.